src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,43 +2,406 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Diagnostics;
-using System.Runtime.CompilerServices;
-using System.Runtime.InteropServices;
 using Internal.Runtime.CompilerServices;
 
 namespace System.Runtime.Intrinsics
 {
-    [Intrinsic]
-    [DebuggerDisplay("{DisplayString,nq}")]
-    [DebuggerTypeProxy(typeof(Vector128DebugView<>))]
-    [StructLayout(LayoutKind.Sequential, Size = 16)]
-    public readonly struct Vector128<T> where T : struct
+    public abstract class Vector128
```

Comment:
This matches what we did with `System.Collections.Immutable` (having a separate, non-generic class, with the same name that holds the `Create` methods).

I've requested another API review (cc. @terrajobst) to ensure that this is what we want to do. One of the alternatives being just exposing these as constructors on `Vector128<T>` and implicitly converting to the requested type. I opted not to go this route as it would lead to situations like `Vector128<float> x = new Vector128<float>(int, int, int, int)` or `Vector128<float> x = new Vector128<float>(double, double)`, which may be confusing (since it would create a vector based on the input type and then bitcast to the target type).

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.tuple is another example of this API pattern.